### DISCUSSION TITLE
Architecture

### DISCUSSION THREAD

**Discussion by evenstensberg** [evenstensberg/structe#8] [Ideas]
Let's figure out the best way to serve front end frameworks and front end infrastructure

**Comment by evenstensberg** [comment]
1. I'm in favor for writing the core library in rust, with the CLI in vlang or nodejs. 
2. The CLI should export a schema for the core library to use.

--- 
**If Rust is the way to go, we have no API entrypoint if developers want to interact with this through code, so that might be a problem.**
---

To ingest tools, we need a context class/namespace that is generated through an interator pattern, where the iterator will ingest tool by tool and turn it into a configuration file.

We also need a translation layer for bundlers, and for libraries we need a translation layer for the front-end library api.

**Reply by ZeroGDrive** [reply]
Is there any specific reason for choosing Rust ? and why vlang for the CLI ? it's quite a new language.
I support using Rust since pretty much all new frontend tools are being built with it. But I want to know if there is any particular feature we're interested in.

**Reply by evenstensberg** [reply]
Mostly performance reasons and memory-safety

**Comment by evenstensberg** [comment]
If we adjust our (internal) architecture, we can have one rust create per underlying functionality. This way, we can also make structe have a programatic tool
